src/frontend/test_runner/tests/testdata/basic_query_1.yaml

```diff
@@ -145,3 +145,27 @@
     BatchSimpleAgg { aggs: [count] }
       BatchExchange { order: [], dist: Single }
         BatchScan { table: t, columns: [] }
+- sql: |
+    /* test boolean expression common factor extraction */
+    create table t (v1 Boolean, v2 Boolean, v3 Boolean);
+    select * from t where v1 AND v2 AND ((v1 AND v2) OR (v2 AND v3));
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchFilter { predicate: $0 AND $1 AND ($0 OR $2) }
+        BatchScan { table: t, columns: [v1, v2, v3] }
+- sql: |
+    /* test boolean expression simplification */
+    create table t (v1 Boolean, v2 Boolean, v3 Boolean);
+    select * from t where v1 AND NOT(v1 OR v2 Or NOT(v1 AND v2 AND true));
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchFilter { predicate: $0 AND Not($0) AND Not($1) AND $1 }
+        BatchScan { table: t, columns: [v1, v2, v3] }
+- sql: |
+    /* test boolean expression simplification */
+    create table t (v1 Boolean, v2 Boolean);
+    select * from t where (v1 AND v2) OR (v1 AND v2);
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchFilter { predicate: $0 AND $1 AND (true:Boolean OR true:Boolean) }
```

Comment:
Seems that an extra `True | True` is generated... :thinking:

Reply:
Maybe constant folding, not pushdown, and factorization should be applied in a single pass to avoid this kind of situation. 

But maybe it is kind of out of the scope of this PR. The correctness is not affected anyway.

Reply:
Given the fact that factorization will introduce new constants, we can apply a boolean constant fold pass before returning.